### PR TITLE
docs: add reference and getting-started docs

### PR DESCRIPTION
**Pull request by jaredallard**
## What this PR does / why we need it

This PR adds basic documentation around getting started and reference docs.

## Jira ID

[DTSS-1378]

## Notes for your reviewers

[DTSS-1378]: https://outreach-io.atlassian.net/browse/DTSS-1378?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: docs/reference-docs/module-hooks.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+{{- end }}
+];
+```
+
+Module A can hook into that section by doing the following:
+
+```tpl
+// Don't actually generate this file
+{{- file.Skip "passes arguments to module b" }}
+{{- stencil.AddToModuleHook "module-b" (list "my-deployment.jsonnet") }}
```
Comment (george-e-shaw-iv) [inline docs/reference-docs/module-hooks.md:25]: where does `"module-b"` get defined?

Comment (jaredallard) [reply docs/reference-docs/module-hooks.md:25]: Right above :sob:

File: docs/reference-docs/templates.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
+# Templates
+
+A template is a `<filename>.ext.tpl` in a stencil module.
```
Comment (george-e-shaw-iv) [inline docs/reference-docs/templates.md:3]:
```suggestion
A template is a `<filename>.<extension>.tpl` in a stencil module. 
```
i think thats more clear. could be interpreted that `.ext.tpl` is always necessary the other way

File: docs/reference-docs/templates.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+**Mode**: By default the mode of the file is taken from template's mode, this can be changed with `file.SetMode`.
+
+## File/Stencil Functions
+
+The `file` and `stencil` functions are scoped to a given template/module (depending on the context) that's currently
+being rendered by stencil.
+
+A complete list of all functions available to a template can be found in pkg.go.dev:
+
+ * [`file`](https://pkg.go.dev/github.com/getoutreach/stencil@v1.5.0/internal/codegen#TplFile)
```
Comment (george-e-shaw-iv) [inline docs/reference-docs/templates.md:18]: do you have to tie these paths to a version?

Comment (jaredallard) [reply docs/reference-docs/templates.md:18]: Nope, we could remove the versions

**Review by george-e-shaw-iv — APPROVED** [review]

**Pull request merged**